lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -336,6 +336,38 @@ defmodule ExUnit.AssertionsTest do
       "Expected exception SyntaxError but got FunctionClauseError (no function clause matching in :lists.flatten/1)" = error.message
   end
 
+  test "assert raise comparing messages (for equality)" do
+    assert_raise RuntimeError, "foo", fn ->
+      raise RuntimeError, "bar"
+    end
+  rescue
+    error in [ExUnit.AssertionError] ->
+      expected = String.rstrip("""
+      Wrong message for RuntimeError
```

Comment:
Maybe go using `<>` like in every other test? [For example](https://github.com/whatyouhide/elixir/blob/better-assert-raise-messages/lib/ex_unit/test/ex_unit/assertions_test.exs#L157-L161).